solver/llbsolver/solver.go

```diff
@@ -142,6 +143,20 @@ func (s *Solver) Solve(ctx context.Context, id string, sessionID string, req fro
 	res.EachRef(func(ref solver.ResultProxy) error {
 		eg.Go(func() error {
 			_, err := ref.Result(ctx2)
+			if err != nil {
+				// Also release any results referenced by exec errors.
+				var ee *errdefs.ExecError
+				if errors.As(err, &ee) {
+					ee.EachRef(func(res solver.Result) error {
+
+						workerRef, ok := res.Sys().(*worker.WorkerRef)
+						if !ok {
+							return nil
```

Comment:
Why are we returning `nil` here?  It seems like we should return `err` from the `ref.Result`?

Reply:
The `err` from `ref.Result` is returned on line 160, and we're currently dropping the error from `ee.EachRef` like how we do in the defer on line 135 when dealing with releasing results, but perhaps I shouldn't be following that anti-pattern?

Reply:
I see, I missed this was in the `EachRef` callback, it seems fine as it.